lib/livebook/session.ex

```diff
@@ -1442,6 +1470,15 @@ defmodule Livebook.Session do
     state
   end
 
+  defp after_operation(state, _prev_state, {:delete_secret, _client_id, secret_name}) do
+    if Runtime.connected?(state.data.runtime), do: delete_runtime_secrets(state, [secret_name])
+    state
+  end
+
+  defp after_operation(state, _prev_state, {:put_notebook_secret, _client_id, _secret}) do
+    notify_update(state)
+  end
```

Comment:
We don't need `notify_update`, since this operation doesn't affect the `%Session{}` struct. That said, we should remove the notebook secret part altogether :)